parser/src/lib.rs

```diff
@@ -517,7 +517,7 @@ impl<'input, I, Id, F> ParserEnv<I, F>
                                   sep_end_by(self.expr(), token(Token::Comma)))
                          .map(|exprs| {
                              loc(Expr::Array(Array {
-                                 id: self.empty_id.clone(),
+                                 typ: Type::hole(),
```

Comment:
Store a `Type::hole()` in the parser instance instead and clone that (just like `empty_id`).